### PR TITLE
(EAI-143): Update Atlas Vector Search index definitions and query syntax

### PR DESCRIPTION
**Pull request by mongodben**
Jira: https://jira.mongodb.org/browse/EAI-143

## Changes

- Update Atlas Vector Search index definition and query syntax
- Upgrade Node driver to support this
- Add a staging kanopy cronjob to keep test DB embedded content up to date (was running into issues w/ the new search query)
- Add new index definitions to the docs
- misc little fixes to accommodate package upgrades 

## Notes

- Also added the new Atlas Vector Search index to all the clusters with the name `vector_index`. When these changes are deployed to staging/prod they should *just work*. once they're validated to work, we can delete the old indexes 
- already deployed to staging, where it works.

### CODE REVIEW

File: packages/mongodb-chatbot-ui/src/ModalView.tsx [hunk 1]
```diff
@@ -143,6 +143,7 @@ export function ModalView(props: ModalViewProps) {
       size="large"
       initialFocus={`#${ActiveInputBarId}`}
       shouldClose={shouldClose}
+      placeholder=""
```
Comment (mongodben) [inline packages/mongodb-chatbot-ui/src/ModalView.tsx:146]: @nlarew when i updated dependencies for the project (deleting node_modules + package-lock.json), i had to add the `placeholder` property here. this seems to work w/o visual regression based on manual testing, but could you take a look at this before merge?

File: packages/ingest-mongodb-public/environments/staging.yml [hunk 2]
```diff
@@ -30,6 +30,33 @@ cronJobs:
       limits:
         cpu: 500m
         memory: 5Gi
+  - name: ingest-content-test
```
Comment (nlarew) [inline packages/ingest-mongodb-public/environments/staging.yml:33]: Do we still need this?

File: packages/mongodb-chatbot-server/src/routes/conversations/addMessageToConversation.test.ts [hunk 3]
```diff
@@ -328,9 +328,9 @@ describe("POST /conversations/:conversationId/messages", () => {
       );
     });
     test("Should respond with 200 and static response if no vector search content for user message", async () => {
-      // Lorem ipsum in Burmese (apparently), if you're curious what this is. Thanks ChatGPT.
+      // Random mix of amharic, burmese, and georgian interpolated with emojis (if you're curious)
```
Comment (nlarew) [inline packages/mongodb-chatbot-server/src/routes/conversations/addMessageToConversation.test.ts:331]: For every ten likes, I'll ask GPT to make this nonsense message even more arcane

File: packages/mongodb-chatbot-server/src/services/ChatLlm.ts [hunk 4]
```diff
@@ -4,10 +4,10 @@
  * from that. This interface could still work with non OpenAI providers if they
  * implement the same interface.
  */
-import { ChatMessage, ChatCompletions } from "@azure/openai";
+import { ChatCompletions } from "@azure/openai";
 export type OpenAiMessageRole = "system" | "assistant" | "user";
 
-export interface OpenAiChatMessage extends ChatMessage {
+export interface OpenAiChatMessage {
```
Comment (nlarew) [inline packages/mongodb-chatbot-server/src/services/ChatLlm.ts:10]: Given that this is an OpenAI-specific interface, why not inherit the OpenAI `ChatMessage` interface?

Comment (mongodben) [reply packages/mongodb-chatbot-server/src/services/ChatLlm.ts:10]: they removed it from the latest version of the SDK

Comment (mongodben) [reply packages/mongodb-chatbot-server/src/services/ChatLlm.ts:10]: we were getting a typescript compile err b/c the sdk doesn't export the type any more

File: packages/mongodb-chatbot-ui/src/ModalView.tsx (hunk 1 above)
Comment (nlarew) [reply packages/mongodb-chatbot-ui/src/ModalView.tsx:146]: Strange what error do you get if you remove it? Modal doesn't take a `placeholder` prop (I'm not really sure what that would even mean)

I can't reproduce but sounds like something wacky is happening

Comment (mongodben) [reply packages/mongodb-chatbot-ui/src/ModalView.tsx:146]: try to 1) check out this branch 2) `rm -rf node_modules package-lock.json` 3) npm i lerna && npm run bootstrap

File: packages/mongodb-rag-core/src/MongoDbEmbeddedContentStore.ts [hunk 5]
```diff
@@ -80,40 +80,43 @@ export function makeMongoDbEmbeddedContentStore({
...
-              knnBeta: {
-                vector,
-                path,
-                k,
-                filter,
-              },
+              queryVector: vector,
+              path,
+              limit: k,
+              numCandidates: numCandidates ?? k * 15,
```
Comment (nlarew) [inline packages/mongodb-rag-core/src/MongoDbEmbeddedContentStore.ts:112]: From [`$vectorSearch` docs](https://www.mongodb.com/docs/atlas/atlas-vector-search/vector-search-stage/):
> we recommend a ratio of ten to twenty nearest neighbors

Just checking, did you choose 15 to split the difference here or is there other guidance that points at 15?

Comment (mongodben) [reply packages/mongodb-rag-core/src/MongoDbEmbeddedContentStore.ts:112]: > did you choose 15 to split the difference here

exactly this

---

and user can override this default behavior by providing `numCandidates`

File: packages/mongodb-chatbot-ui/src/ModalView.tsx (hunk 1 above)
Comment (nlarew) [reply packages/mongodb-chatbot-ui/src/ModalView.tsx]: Fixed (I think) in the latest commit by upgrading the version of `@leafygreen-ui/modal`

**Review by nlarew — APPROVED** [review]
🚢

**Pull request merged**